ui/component.js

```diff
@@ -2904,6 +2970,205 @@ var Component = exports.Component = Target.specialize(/** @lends Component.proto
         }
     },
 
+    // Drag & Drop operations
+
+    /**
+     * Register a component for beeing a dragging source.
+     */
+    registerForDragSource: {
+        value: function () {
+            this.dragManager.registerForDragSource(this);
+            this.__isDragSource = true;
+            this.classList.add("montage-drag-source");
+        }
+    },
+
+    /**
+     * unregister a component for beeing a dragging source.
+     */
+    unregisterForDragSource: {
+        value: function () {
+            this.dragManager.unregisterForDragSource(this);
+            this.classList.remove("montage-drag-source");
+        }
+    },
+
+    /**
+     * Register a component for beeing a drag destination.
+     */
+    registerForDragDestination: {
+        value: function () {
+            this.dragManager.registerForDragDestination(this);
+            this.__isDragDestination = true;
+            this.classList.add("montage-drag-destination");
+        }
+    },
+
+    /**
+     * Unregister a component for beeing a drag destination.
+     */
+    unregisterForDragDestination: {
+        value: function () {
+            this.dragManager.unregisterForDragDestination(this);
+            this.classList.remove("montage-drag-destination");
+        }
+    },
+
+    /**
+     * Called on the source when the drag operation starts.
+     */
+    _beginDraggingOperation: {
+        value: function (draggingOperationInfo) {
+            if (typeof this.beginDraggingOperation === "function") {
+                this.beginDraggingOperation(draggingOperationInfo);
+            }
+        }
+    },
+
+    /**
+     * Called on the source when the dragged image is moving.
+     */
+    _updateDraggingOperation: {
+        value: function (draggingOperationInfo) {
+            if (typeof this.updateDraggingOperation === "function") {
+                this.updateDraggingOperation(draggingOperationInfo);
+            }
+        }
+    },
+
+    /**
+     * Called on the source when the drag operation ends.
+     */
+    _endDraggingOperation: {
+        value: function (draggingOperationInfo) {
+            if (typeof this.endDraggingOperation === "function") {
+                this.endDraggingOperation(draggingOperationInfo);
+            }
+        }
+    },
+
+    /**
+     * Called when the drag operation start,
+     * allowing the receivers to agree to or refuse the drag operation.
+     * By default all receivers will not accept any drag operation.
+     * The method `draggingStarted` should return a boolean value.
+     */
+    _draggingStarted: {
```

Comment:
@marchant Needs to find a matching event name, to tell a potential destination that a drag started